components/Header/index.js

```diff
@@ -1,16 +1,102 @@
 import styled from "styled-components";
+import { Spin as Hamburger } from "hamburger-react";
+import Link from "next/link";
+import { useState } from "react";
 
 export default function Header() {
-  return <StyledHeader>TrailBlazer</StyledHeader>;
+  const [isOpen, setIsOpen] = useState(false);
+
+  return (
+    <>
+      <StyledHeader>
+        <StyledHeading>TrailBlazer</StyledHeading>
+        <StyledMenu>
+          <Hamburger
+            toggled={isOpen}
+            toggle={setIsOpen}
+            duration={0.6}
+            label="Show Menu"
+          />
+        </StyledMenu>
+      </StyledHeader>
+      {isOpen ? (
+        <HamburgerMenu>
+          <MenuLink onClick={() => setIsOpen(false)} href="/bookmark">
+            My Bookmarks
+          </MenuLink>
+          <MenuLink onClick={() => setIsOpen(false)} href="/">
+            Home
+          </MenuLink>
+          <MenuLink onClick={() => setIsOpen(false)} href="/spotlight">
+            Inspire me!
+          </MenuLink>
+          <MenuLink onClick={() => setIsOpen(false)} href="/createActivity">
+            Add new Activity
+          </MenuLink>
+        </HamburgerMenu>
+      ) : null}
```

Comment:
This works completely fine, however, if the else case is `null`, you could also write it like this:

```suggestion
      {isOpen && (
        <HamburgerMenu>
          <MenuLink onClick={() => setIsOpen(false)} href="/bookmark">
            My Bookmarks
          </MenuLink>
          <MenuLink onClick={() => setIsOpen(false)} href="/">
            Home
          </MenuLink>
          <MenuLink onClick={() => setIsOpen(false)} href="/spotlight">
            Inspire me!
          </MenuLink>
          <MenuLink onClick={() => setIsOpen(false)} href="/createActivity">
            Add new Activity
          </MenuLink>
        </HamburgerMenu>
      )}
```